### ISSUE TITLE
Qualitative metrics can be misleading with `beta_diversity`

### ISSUE THREAD

**Issue by wasade** [biocore/scikit-bio#1549]
When calling a qualitative metric, such as Jaccard, the input counts data are not automatically cast to presence/absence. The user may expect that to happen, and this can produce misleading results.

This issue was first opened on qiime2/q2-diversity#189. 

An example of the issue can be seen below. @gregcaporaso suggests two possible resolutions: (1) implementing the same steps as sklearn, or (2) updating skbio to call `sklearn.metrics.pairwise_distances` (plus new tests in skbio to ensure that we know if something changes in sklearn). 

Two near term work around options are to either cast input data to presence/absence prior to executing a qualitative metric, or to use `sklearn.metrics.pairwise_distances` as shown below.

The fast UniFrac methods are not impacted (e.g., 'unweighted_unifrac', and 'weighted_unifrac').

```python
import skbio
import sklearn
import numpy as np

>>> a = np.array([
    [42, 0, 37, 99, 1],
    [12, 1, 22, 88, 0],
    [25, 3, 23, 86, 0],
    [0, 0, 87, 12, 0]]
)

>>> dm1 = skbio.diversity.beta_diversity('jaccard', a, ids=['4ac2', 'e375', '4gd8', '9872'])
>>> dm2 = skbio.diversity.beta_diversity('jaccard', a > 0, ids=['4ac2', 'e375', '4gd8', '9872'])
>>> dm1 == dm2
False

>>> dm1 = skbio.diversity.beta_diversity('jaccard', a, ids=['4ac2', 'e375', '4gd8', '9872'], pairwise_func=sklearn.metrics.pairwise_distances)
>>> dm2 = skbio.diversity.beta_diversity('jaccard', a > 0, ids=['4ac2', 'e375', '4gd8', '9872'], pairwise_func=sklearn.metrics.pairwise_distances)
>>> dm1 == dm2
True
```

**Comment by gregcaporaso** [comment]
Thanks @wasade, and thanks for originally tracking this down @mahermassoud. 

I would be in favor of option (2) above, with new unit tests. We previously hadn't done that because we didn't want to add sklearn as a dependency, but it's been a core QIIME 2 dependency for a while now, and it hasn't caused any problems (in fact, it saved us from this one) so I don't see a problem with making it a dependency of skbio. Any objections to that?

**Comment by wasade** [comment]
I agree that (2) is preferable, and I think that laying a path toward integration with sklearn opens a lot of exciting doors.

**Comment by lkursell** [comment]
Thanks @wasade, I rediscovered this bug this morning on my unit tests and thought I was loosing my mind. Thanks for the fix.

**Comment by wasade** [comment]
Don't worry, your mind is still intact 😄

**Comment by tanaes** [comment]
...aaaand I just figured out a very circuitous way to learn that I need to update my skbio library 😅
